Student-Unity-frontend/src/app/services/student-task.service.ts

```diff
@@ -0,0 +1,80 @@
+import { SignInOutService } from './sign-in-out.service';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { Task } from '../models/Task';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StudentTaskService {
+
+  private url = environment.baseUrl + "/myTasks";
+
+  constructor(private http: HttpClient, private userService: SignInOutService) { }
+
+  public getStudentTasks(): Observable<HttpResponse<Task[]>> {
+    return this.http.get<Task[]>(this.url + '/all', {
+      observe: 'response',
+      params: {
+        sessionId: this.userService.getSignedInUserSessionID(),
+      },
+      responseType: 'json',
+    });
+  }
+
+  public getSubscribedCourses(): Observable<HttpResponse<string[]>> {
+    return this.http.get<string[]>(this.url + '/allSubscribedCourses', {
+      observe: 'response',
+      params: {
+        sessionId: this.userService.getSignedInUserSessionID(),
+      },
+      responseType: 'json'
+    })
+  }
+
+  public filterToDoTasks(courseCode: string): Observable<HttpResponse<Task[]>> {
+    return this.http.get<Task[]>(this.url + '/filterToDoTasks', {
+      observe: 'response',
+      params: {
+        sessionId: this.userService.getSignedInUserSessionID(),
+        courseCode: courseCode
+      },
+      responseType: 'json',
+    });
+  }
+  
+  public filterDoneTasks(courseCode: string): Observable<HttpResponse<Task[]>> {
+    return this.http.get<Task[]>(this.url + '/filterDoneTasks', {
+      observe: 'response',
+      params: {
+        sessionId: this.userService.getSignedInUserSessionID(),
+        courseCode: courseCode
+      },
+      responseType: 'json',
+    });
+  }
```

Comment:
Good idea to sent the requests for filter todo tasks throw two end points so that we make less logic in the front end and to provide more usability to the user